### PR TITLE
Bl 102 create messages component

### PR DESCRIPTION
**Pull request by jose-mercado**
Co-authored-by: Ahmad Wahidy shah.wahidy@gmail.com
Co-authored-by: William Lee leewp8@gmail.com
Co-authored-by: Spencer Longhurst spencersean12@gmail.com

## Description

Created Student messages component based on the figma design and made sure that it was linked up to the student dashboard. Added list of conversations next to actual conversation the student it having. 

Fixes # BL-102
Ticket link: https://bloomtechlabs.atlassian.net/browse/BL-102

## Loom Video

[Loom Video](https://www.loom.com/share/55a5e1fd3bce4bc9ba1327f31fdb9a3c)
<a href="https://www.loom.com/share/55a5e1fd3bce4bc9ba1327f31fdb9a3c">
    <p>Brave - [BL-102] Create "Messages" Component - Jira - Brave - 22 June 2022 - Watch Video</p>
    
  </a>

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

[BL-102]: https://bloomtechlabs.atlassian.net/browse/BL-102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by kevkenshi-max — DISMISSED** [review]
Nicely done with building an essential component. Messaging.

**Review by DawsonReschke — DISMISSED** [review]
Good work on completing the messages component, heads up there are merge conflicts you might want to resolve.

**Comment by jose-mercado** [comment]
> Good work on completing the messages component, heads up there are merge conflicts you might want to resolve.

Thank you! got it fixed

**Review by kylerostler — APPROVED** [review]
I think there is some sort of issue with your loom video link, but the code looks great! Component is built cleanly, there are some pieces of commented code not sure if that was intentional. All in all the components look good and I love the fact that you got this feature built, very cool.

**Review by Ogden-R — APPROVED** [review]
great job making the page look so much like the figma layout of it. great example of making a concept into reality. looks great and very user friendly.  and i like the star wars easter egg in here too!

**Review by brandib222 — DISMISSED** [review]
Wow, you guys really went for it! I would take out some of the comments, but otherwise, it looks well organized. Great job!

**Review by anastasia-lapteva — DISMISSED** [review]
Hello Jose, Ahmad, William, and Spencer,

Your team did a great job building this component! Also thank you for demonstrating the comment feature in your Loom video, it was cool to see in action. I noticed in your code for ActiveMessage.js, MessageList.js, MessagesContainer.js, and ReplyEditor.js you left some commented out imports at the top of the pages. These imports are accompanied with "possible location for styling component below". I'm not sure if that means the styling component may be located at those import locations, or if you would like to add a styling component to those locations in the future. Regardless, if the styling does not currently exist perhaps this commented-out code is not needed at this time.

File: src/components/pages/StudentHome/index.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import Banner from '../../common/Banner';
 import StudentSideBar from './StudentSidebar';
 import StudentDashboard from './StudentDashboard';
 import '../../../styles/ParentStyles/index.less';
-import InstructorQuestions from './InstructorQuestions.js';
+// import InstructorQuestions from './InstructorQuestions.js';
```
Comment (Michael-Subbarao) [inline src/components/pages/StudentHome/index.js:8]: Good job commenting out unused imports to avoid warnings.

**Review by Michael-Subbarao — APPROVED** [review]
The code and the component look clean. Great work!

**Pull request merged**